src/main/java/com/redhat/cloud/notifications/routers/UserConfigService.java

```diff
@@ -159,45 +137,15 @@
-    private Uni<SettingsValues> oldGetSettingsValueForUser(String account, String username) {
-        return Uni.createFrom().deferred(() -> {
-            final SettingsValues values = new SettingsValues();
-
-            return bundleResources.getBundles()
-                    .onItem().transformToMulti(bundle -> {
-                        BundleSettingsValue bundleSettingsValue = new BundleSettingsValue();
-                        bundleSettingsValue.displayName = bundle.getDisplayName();
-                        values.bundles.put(bundle.getName(), bundleSettingsValue);
-                        return applicationResources.getApplications(bundle.getName())
-                                .onItem().transformToMulti(application -> {
-                                    ApplicationSettingsValue applicationSettingsValue = new ApplicationSettingsValue();
-                                    applicationSettingsValue.displayName = application.getDisplayName();
-                                    values.bundles.get(bundle.getName()).applications.put(application.getName(), applicationSettingsValue);
-                                    return Multi.createFrom().items(EmailSubscriptionType.values())
-                                            .onItem().transformToMulti(emailSubscriptionType -> {
-                                                values.bundles.get(bundle.getName()).applications.get(application.getName()).notifications.put(emailSubscriptionType, false);
-                                                return Multi.createFrom().empty();
-                                            }).merge();
-                                }).merge();
-                    }).concatenate().collect().asList().onItem().transformToMulti(objects -> emailSubscriptionResources.getEmailSubscriptionsForUser(account, username))
-                    .onItem().transform(emailSubscription -> {
-                        values.bundles.get(emailSubscription.getBundleName()).applications.get(emailSubscription.getApplicationName()).notifications.put(emailSubscription.getType(), true);
-                        return emailSubscription;
-                    }).collect().asList().onItem().transform(emailSubscriptions -> values);
-        });
-    }
-
     /**
      * Pulls the user settings values of an user across all the know applications of a bundle
      * */
     private Uni<SettingsValues> getSettingsValueForUser(String account, String username, String bundleName) {
         return Uni.createFrom().deferred(() -> {
             final SettingsValues values = new SettingsValues();
 
-            return bundleResources.getBundles()
-                    // Todo: Create a method that gets the bundle by name
-                    .filter(bundle -> bundle.getName().equals(bundleName))
+            return bundleResources.getBundle(bundleName)
+                    .onItem().ifNull().failWith(() -> new BadRequestException("bundleName must have a value"))
```

Comment:
The `BadRequestException` will also be thrown if `bundleName` has a value but it is unknown.

Reply:
Ah, good catch. I'll update.